src/transformers/models/depth_pro/configuration_depth_pro.py

```diff
@@ -0,0 +1,245 @@
+        # make sure these parameters of backbone model are same as DepthPro
+        matching_config = [
+            "hidden_size",
+            "num_channels",
+            "num_hidden_layers",
+            "num_attention_heads",
+            "_attn_implementation",
+        ]
+        matching_config_dict = {k: getattr(self, k) for k in matching_config}
+
+        if backbone_config is None:
+            # use Dinov2 config by default
+            logger.info("Initializing the config with the default Dinov2 backbone.")
+            backbone_config = Dinov2Config(
+                patch_size=16,
+                image_size=self.patch_size,
+                **matching_config_dict,
+            )
+
+        elif isinstance(backbone_config, dict):
+            assert backbone_config.get("model_type") is not None
+            logger.info(f"Initializing the config with a {backbone_config.get('model_type')} backbone.")
+            backbone_config.update(
+                {
+                    "image_size": self.patch_size,
+                    **matching_config_dict,
+                }
+            )
+            config_class = CONFIG_MAPPING[backbone_config.get("model_type")]
+            backbone_config = config_class.from_dict(backbone_config)
+
+        elif isinstance(backbone_config, PretrainedConfig):
+            backbone_config = backbone_config
+
+        else:
+            raise ValueError(
+                f"backbone_config must be a dictionary or a `PretrainedConfig`, got {backbone_config.__class__}."
+            )
+
+        # validate the config compatibility between DepthPro and the backbone model
+        if self.patch_size != backbone_config.image_size:
+            # patches of input image are created of size patch_size x patch_size
+            # then these patches are given to the backbone model as input images
+            raise ValueError(
+                f"patch_size={self.patch_size} should be equal to backbone_config.image_size={backbone_config.image_size}."
+            )
+        for key in matching_config:
+            config_value = getattr(self, key)
+            backbone_config_value = getattr(backbone_config, key)
+            if config_value != backbone_config_value:
+                raise ValueError(
+                    f"{key}={config_value} should be equal to backbone_config.{key}={backbone_config_value}."
+                )
+
+        return backbone_config
+
+    def to_dict(self):
+        """
+        Serializes this instance to a Python dictionary. Override the default [`~PretrainedConfig.to_dict`]. Returns:
+            `Dict[str, any]`: Dictionary of all the attributes that make up this configuration instance,
+        """
+        output = copy.deepcopy(self.__dict__)
+
+        if output["backbone_config"] is not None:
+            output["backbone_config"] = self.backbone_config.to_dict()
+
+        output["model_type"] = self.__class__.model_type
+        return output
```

Comment:
One more comment, please see how the composite models and configs are built, for example CLIP/SigLIP. You should have one config inside another, and not flatten it, for consistency with other models.